addons/mail/static/tests/messaging_menu/messaging_menu_tests.js

```diff
@@ -1044,18 +1044,30 @@ QUnit.test("failure notifications are shown before channel preview", async () =>
 QUnit.test("messaging menu should show new needaction messages from chatter", async () => {
     const pyEnv = await startServer();
     const partnerId = pyEnv["res.partner"].create({ name: "Frodo Baggins" });
-    await start();
+    const { env } = await start();
     await click(".o_menu_systray i[aria-label='Messages']");
-    await contains(".o-mail-NotificationItem-text", { count: 0, text: "@Mitchell Admin" });
+    await contains(".o-mail-NotificationItem-text", {
+        count: 0,
+        text: "Frodo Baggins: @Mitchel Admin",
+    });
     // simulate receiving a new needaction message
-    pyEnv["bus.bus"]._sendone(pyEnv.currentPartner, "mail.message/inbox", {
+    const messageId = pyEnv["mail.message"].create({
         author_id: partnerId,
-        body: "@Mitchell Admin",
-        id: 100,
-        needaction_partner_ids: [pyEnv.currentPartnerId],
+        body: "@Mitchel Admin",
+        needaction: true,
         model: "res.partner",
         res_id: partnerId,
-        record_name: "Frodo Baggins",
+        needaction_partner_ids: [pyEnv.currentPartnerId],
+    });
+    pyEnv["mail.notification"].create({
+        mail_message_id: messageId,
+        notification_status: "sent",
+        notification_type: "inbox",
+        res_partner_id: pyEnv.currentPartnerId,
     });
-    await contains(".o-mail-NotificationItem-text", { text: "@Mitchell Admin" });
+    const [formattedMessage] = await env.services.orm.call("mail.message", "message_format", [
+        [messageId],
+    ]);
+    pyEnv["bus.bus"]._sendone(pyEnv.currentPartner, "mail.message/inbox", formattedMessage);
```

Comment:
This could indeed be a better test. However it would take time to do (probably some missing code in mock server, mock server might be too limited to test another user triggering this by sending a new message, etc.) and I think it's better to keep the test as is so that I focus on cleaning discuss models as soon as possible.

Tests help giving a certain level of confidence on the code. If adapting this test now takes too much time (very likely), this is a waste of time for me. I prefer to consider cleaning tests in another task/PR when this becomes a higher priority.